Add Token Credential Samples with DTFx.AzureStorage v1 and v2 (#1118)
Azure/durabletask: samples/ManagedIdentitySample/DTFx.AzureStorage v1.x/ReadMe.md

# Token Credential Sample

This sample demonstrates how to configure a Identity based connection when using DurableTask.AzureStorage v1.x as the orchestration provider for a Durable Task project.

> Note: 
> Identity based connection **is not supported** with .NET framework 4.x with DurableTask.AzureStorage v1.x

## Configuration Prerequisites

Before running this sample, you must

> 1. Create a new Azure Storage account, or reuse an existing one

Reviewer: ```suggestion
1. Create a new Azure Storage account or reuse an existing one
```

Removing the comma

> 2. Create your identity in Azure Portal. For Managed Identity, detailed instructions can be found [here](https://learn.microsoft.com/azure/app-service/overview-managed-identity?tabs=portal%2Chttp). For Client Application, instructions can be found [here](https://learn.microsoft.com/azure/healthcare-apis/register-application). 

Reviewer 2: That second link (https://learn.microsoft.com/azure/healthcare-apis/register-application) is a bit surprising to me, since it's for "healthcare apis". Do we link to this in our other mslearn docs or how did you find this link?

Reviewer 2: Here's my tip -  link to the "official" docs here instead: https://learn.microsoft.com/en-us/entra/identity-platform/quickstart-register-app?tabs=certificate and, **if you want, *you can say that the other link (the health care one) gives a simpler step-by-step explanation if

> 2. Create your identity in Azure Portal. For Managed Identity, detailed instructions can be found [here](https://learn.microsoft.com/azure/app-service/overview-managed-identity?tabs=portal%2Chttp). For Client Application, instructions can be found [here](https://learn.microsoft.com/azure/healthcare-apis/register-application). 

Reviewer: ```suggestion
2. Create your identity in the Azure Portal. For Managed Identity, detailed instructions can be found [here](https://learn.microsoft.com/azure/app-service/overview-managed-identity?tabs=portal%2Chttp). For Client Application, instructions can be found [here](https://learn.microsoft.com/azure/healthcare-apis/register-application). 
```

> 2. Create your identity in Azure Portal. For Managed Identity, detailed instructions can be found [here](https://learn.microsoft.com/en-us/azure/app-service/overview-managed-identity?tabs=portal%2Chttp). For Client Application, instructions can be found [here](https://learn.microsoft.com/en-us/azure/healthcare-apis/register-application). 

Reviewer 3: Please remove "en-us" from the links.

Reviewer 3: Do we need to register the EntraID for managed identity to work? EntraID seems to be another way of authenticating?

Author: Entra Id is a platform that contains all kinds of identity and access. When we register an identity in Azure, it's already under the Entra Id platform. I think we don't need to add this item here in case others might feel confused.

Reviewer 2: If it helps, Entra is the new name for "Azure Active Directory"/ AAD

Reviewer 3: > Thanks for review! Entra Id is a platform that contains all kinds of identity and access. When we register an identity in Azure, it's already under the Entra Id platform. I think we don't need to add this item here in case others might feel confused.

I was confused, so if this is not needed, we can remove the second sentence.

Author: I remove the second link and use our durable managed identity document link for instructions about how to create identity.

> 3. Assign Role-based Access Controls (RBAC) of the storage account in step 1 to the identity created in step 2 with [these instructions](https://learn.microsoft.com/azure/role-based-access-control/role-assignments-portal-managed-identity#Overview).  

Reviewer: ```suggestion
3. Assign Role-based Access Controls (RBAC) of the storage account created in step 1 to the identity created in step 2 with [these instructions](https://learn.microsoft.com/azure/role-based-access-control/role-assignments-portal-managed-identity#Overview).  
```

        * Storage Queue Data Contributor
        * Storage Blob Data Contributor
        * Storage Table Data Contributor
4. Add the identity required information to your app's configuration.

> 4. Set `AccountName` the name of the storage account. AccountName can be replaced with Storage Account BlobServiceUri, TableServiceUri and QueueServiceUri. 

Reviewer: ```suggestion
5. Set `AccountName` to the name of the storage account. AccountName can be replaced with Storage Account BlobServiceUri, TableServiceUri and QueueServiceUri. 
```
